### PR TITLE
app.js file

### PR DESCRIPTION
**Pull request by maccrin**
This PR for meal-sharing week1

### CODE REVIEW

File: src/backend/app.js [hunk 1]
```diff
@@ -20,12 +20,35 @@ app.use(express.json());
...
+});
 
 if (process.env.API_PATH) {
   app.use(process.env.API_PATH, router);
 } else {
   throw "API_PATH is not set. Remember to set it in your .env file"
 }
+app.get("/future-meals", async (req, res) => {
+  const rows = await knex.raw("SELECT * FROM `meal` WHERE `when` > NOW()");
+  rows[0].length === 0 ? res.sendStatus(404) : res.send(rows.slice(0, 1))
```
Comment (Divyajg) [inline src/backend/app.js]: Even though there are no meals available, it shouldn't be an error response as it is responding with the correct data.
So the response is Status:200 and result:"There are not meals available".

File: src/backend/app.js [hunk 2]
```diff
@@ -20,12 +20,35 @@ app.use(express.json());
...
+  res.send("Hi friend");
+});
 
 if (process.env.API_PATH) {
   app.use(process.env.API_PATH, router);
 } else {
   throw "API_PATH is not set. Remember to set it in your .env file"
 }
+app.get("/future-meals", async (req, res) => {
+  const rows = await knex.raw("SELECT * FROM `meal` WHERE `when` > NOW()");
```
Comment (Divyajg) [inline src/backend/app.js]: Future meals include the meals that we have today so, `when >= now()` and for Past meals `when < now()`.

**Review by Divyajg — CHANGES_REQUESTED** [review]
Great work. Hope the comments help you understand what I mean. that comments applies wherever it is applicable.

File: src/backend/app.js (hunk 1 above)
Comment (maccrin) [reply src/backend/app.js]: I will modify this ,thank you

File: src/backend/app.js (hunk 2 above)
Comment (maccrin) [reply src/backend/app.js]: 👍🏻

File: src/backend/app.js [hunk 3]
```diff
@@ -30,24 +30,24 @@ if (process.env.API_PATH) {
   throw "API_PATH is not set. Remember to set it in your .env file"
 }
 app.get("/future-meals", async (req, res) => {
-  const rows = await knex.raw("SELECT * FROM `meal` WHERE `when` > NOW()");
-  rows[0].length === 0 ? res.sendStatus(404) : res.send(rows.slice(0, 1))
+  const rows = await knex.raw("SELECT * FROM `meal` WHERE `when` >= NOW()");
+  rows[0].length === 0 ? res.status(200).send(`No meals found with future dates`) : res.send(rows.slice(0, 1))
```
Comment (Divyajg) [inline src/backend/app.js]: For suppose, while data fetching if something went wrong, there will be an error and we should handle that error to convey it to the user.
To handle errors: we can use `try/catch` or `if/else`.
Think about it and let me know if you have any questions.

**Review by Divyajg — COMMENTED** [review]
One step towards improvising.

File: src/backend/app.js (hunk 3 above)
Comment (maccrin) [reply src/backend/app.js]: I got it,we can apply try catch here,then do we need for all queries individually or one for all? Please advice,thanks.

**Review by Divyajg — APPROVED** [review]
I appreciate your openness.
Keep coding.

**Pull request merged**